### PR TITLE
Add doc for CRD version removal

### PR DESCRIPTION
**Pull request by xing-yang**
This PR adds a section on CRD version removal.

### CODE REVIEW

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md [hunk 1]
```diff
@@ -353,6 +353,36 @@ spec:
 {{< /tabs >}}
 
 
+### Version removal
+
+An API version can be removed if there is a newer API version served and stored in the API server for at least 3 Kubernetes releases.
```
Comment (liggitt) [inline content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:358]: There's nothing inherent about 3 releases before removing a version, that's just how the storage API chose to manage the version

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md [hunk 2]
```diff
@@ -353,6 +353,36 @@ spec:
...
+The older API version cannot be dropped from the CRD manifest until data has been migrated to the newer API version for all clusters that served the older version of the CRD and cleared the older version from status.storedVersions.
+
+{{< tabs name="CustomResourceDefinition_versioning_removal" >}}
+{{% tab name="apiextensions.k8s.io/v1" %}}
```
Comment (Sea-n) [inline content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:362]: I think there should be `{{% /tab %}}` and `{{< /tabs >}}`.

**Review by Sea-n — CHANGES_REQUESTED** [review]

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md [hunk 3]
```diff
@@ -353,6 +353,36 @@ spec:
 {{< /tabs >}}
 
 
+### Version removal
+
+An API version can be removed if there is a newer API version served and stored in the API server for at least 3 Kubernetes releases.
+The older API version cannot be dropped from the CRD manifest until data has been migrated to the newer API version for all clusters that served the older version of the CRD and cleared the older version from status.storedVersions.
```
Comment (liggitt) [inline content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:359]:
```suggestion
An older API version cannot be dropped from a CustomResourceDefinition manifest until existing persisted data has been migrated to the newer API version for all clusters that served the older version of the custom resource, and the old version is removed from the `status.storedVersions` of the CustomResourceDefinition.
```

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md (hunk 2 above)
Comment (liggitt) [inline content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:362]: drop the tabs/tab bits, we only need to show v1

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md [hunk 4]
```diff
@@ -353,6 +353,36 @@ spec:
...
+  names:
+    plural: crontabs
+    singular: crontab
+    kind: CronTab
+  scope: Namespaced
+  versions:
+  - name: v1beta1
+    served: false
+    # This indicates the v1beta1 version of the custom resource is no longer served.
+    # API requests to this version receive a not found error in the server response.
```
Comment (liggitt) [inline content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:378]: suggest putting this comment above the `served: false` line

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md [hunk 5]
```diff
@@ -353,6 +353,36 @@ spec:
...
+    schema: ...
+  - name: v1
+    served: true
+    storage: true
```
Comment (liggitt) [inline content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:382]: suggest commenting on this line to make it clear the new served version should be set as the storage version

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md [hunk 6]
```diff
@@ -353,6 +353,36 @@ spec:
 {{< /tabs >}}
 
 
+### Version removal
```
Comment (msau42) [inline content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:356]: I noticed that at the top of the page in the Overview section there already are some steps regarding version removal. It does include setting served to false first, and also links to upgrading existing objects to a new stored version.

Maybe the existing steps are sufficient?

Comment (xing-yang) [reply content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:356]: Hmm, I see those steps in the Overview section.  I'm fine either way.

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md (hunk 2 above)
Comment (xing-yang) [reply content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md]: Removed the tab as it is not needed here.

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md (hunk 1 above)
Comment (xing-yang) [reply content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md]: Removed this line.

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md (hunk 3 above)
Comment (xing-yang) [reply content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md]: Done

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md (hunk 4 above)
Comment (xing-yang) [reply content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:376]: Done

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md (hunk 2 above)
Comment (xing-yang) [reply content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md]: Dropped tabs/tab.

**Comment by Sea-n** [comment]
There are no other issues to be address.
/lgtm

**Pull request merged**